cardinal/server/handler.go

```diff
@@ -116,9 +116,12 @@ func (t *Handler) makeTxHandler(tx transaction.ITransaction) http.HandlerFunc {
 		}
 
 		submitTx := func() uint64 {
-			tick, _ := t.w.AddTransaction(tx.ID(), txVal, sp)
+			tick, id := t.w.AddTransaction(tx.ID(), txVal, sp)
 
-			res, err := json.Marshal("ok")
+			res, err := json.Marshal(ReceiptID{
+				ID:   id.String(),
```

Comment:
Can we make the transaction hash to be keccak256(signedPayload) instead?

Reply:
https://ethereum.stackexchange.com/questions/45648/how-to-calculate-the-assigned-txhash-of-a-transaction

Reply:
I would prefer to not use tick as a part of the receipt ID here because that can't be calculated in advance in Nakama (you don't know which tick your transaction is going to be executed at)

Reply:
I included the tick because it's an input for reading the signer of a persona tag. Returning the tick would make integrating this change into Nakama a little easier.

However, I think with the addition of transaction receipts we should be able to get by without returning the tick.

RE the keccak256, I'm not familiar with that. Would this be the geth implementation: https://pkg.go.dev/github.com/ethereum/go-ethereum/crypto#Keccak256Hash ?

Reply:
Yeap, that's the geth implementation.

We should return the tick as a part of the receipt but not as a part of the id/hash.

Reply:
To be clear, what I'm advocating for here is that we remove the idea of a "Receipt ID" there should only be a "transaction hash" and the receipt should attach the transaction hash.